src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,97 @@
-import { useState } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const requiredFields = [title, imgUrl, imdbId, imdbUrl];
+
+  const enableAddButton = useMemo(() => {
+    return requiredFields.every(field => field.length > 0);
+  }, requiredFields);
+
+  const addMovie = useCallback(
+    (event: React.FormEvent) => {
+      event.preventDefault();
+      onAdd({
+        title,
+        description,
+        imdbId,
+        imgUrl,
+        imdbUrl,
+      });
+      setTitle('');
+      setDescription('');
+      setImdbId('');
+      setImdbUrl('');
+      setImgUrl('');
+      setCount((prevState) => (prevState + 1));
+    }, [{
+      title,
+      description,
+      imdbId,
+      imgUrl,
+      imdbUrl,
+    }, count],
```

Comment:
Object literal will always be different from the previous render.
```suggestion
    }, [
      title,
      description,
      imdbId,
      imgUrl,
      imdbUrl,
      count,
    ],
```